sql/telemetry_derived/clients_daily_histogram_aggregates_v1.sql.py

```diff
@@ -423,16 +438,40 @@ def get_histogram_probes(histogram_type):
             main_summary_histograms[histogram["name"]] = processes
 
     with urllib.request.urlopen(PROBE_INFO_SERVICE) as url:
-        data = json.loads(url.read().decode())
+        data = json.loads(gzip.decompress(url.read()).decode())
         histogram_probes = {
             x.replace("histogram/", "").replace(".", "_").lower()
             for x in data.keys()
             if x.startswith("histogram/")
         }
+
+        bucket_details = {}
         relevant_probes = {
             histogram: process for histogram, process in main_summary_histograms.items() if histogram in histogram_probes
         }
-        return relevant_probes
+        for key in data.keys():
+            if not key.startswith("histogram/"):
+                continue
+
+            channel = 'nightly'
+            if 'nightly' not in data[key]["history"]:
+                channel = 'beta'
+
+                if 'beta' not in data[key]["history"]:
+                    channel = 'release'
+
+            data_details = data[key]["history"][channel][0]["details"]
+            probe = key.replace("histogram/", "").replace(".", "_").lower()
+            bucket_details[probe] = {
+                "n_buckets": int(eval(str(data_details["n_buckets"]))),
```

Comment:
Is `eval` necessary to cast a string to an integer? 

```python
>>> int("5")
5
```

You may also consider using [`ast.literal_eval`](https://docs.python.org/3/library/ast.html#ast.literal_eval), which will evaluate literals and not expressions.

Reply:
Unfortunately, some probes, (e.g. `POPUP_NOTIFICATION_MAINACTION_TRIGGERED_MS`) have values in the probe info service like `80 * 25` for the value of `n_buckets` etc. So they do need to be evaluated as expressions.

Perhaps can add a comment

Reply:
Leaving a comment about this specific case would be great.